File: spring-boot-security-jwt-auth-mongodb-master/src/main/java/com/knf/dev/repository/EmployeeRepository.java
package com.knf.dev.repository;

import java.util.Optional;

import org.springframework.data.mongodb.repository.MongoRepository;

import com.knf.dev.models.Employee;

public interface EmployeeRepository extends MongoRepository<Employee, String> {

	Optional<Employee> findByEmployeename(String employeename);

	Boolean existsByEmployeename(String employeename);

	Boolean existsByEmail(String email);
}

File: kotlin-spring-main/kotlin-springsecurity-jwt/src/main/kotlin/com/knf/dev/demo/kotlinspringsecurityjwt/repository/EmployeeRepository.kt
package com.knf.dev.demo.kotlinspringsecurityjwt.repository

import com.knf.dev.demo.kotlinspringsecurityjwt.model.Employee
import org.springframework.data.mongodb.repository.MongoRepository
import java.util.*


interface EmployeeRepository : MongoRepository<Employee?, String?> {
    fun findByEmployeename(employeename: String?): Optional<Employee?>?
    fun existsByEmployeename(employeename: String?): Boolean?
    fun existsByEmail(email: String?): Boolean?
}